config/config.go

```diff
@@ -487,6 +487,9 @@ type Log struct {
 	EnableSlowLog       AtomicBool `toml:"enable-slow-log" json:"enable-slow-log"`
 	SlowThreshold       uint64     `toml:"slow-threshold" json:"slow-threshold"`
 	RecordPlanInSlowLog uint32     `toml:"record-plan-in-slow-log" json:"record-plan-in-slow-log"`
+
+	// Make tidb panic if write log operation hang in `Timeout` seconds
+	Timeout int `toml:"timeout" json:"timeout"`
```

Comment:
should be ```string``` and use ```parseDuration``` to convert to int. It is friendly for users.

Reply:
+1, @tiancaiamao could you address this comment?

Reply:
Here the unit is seconds and this is a design choice.
It does not support more general duration like 500ms / 1hour etc.
Why?
**To make the log library still efficient after support timeout, I use ticker instead of timer.**

```
func getLockOrBlock() bool {
    select {
        case lock <-struct{}:
        case <-time.After(t):    // object allocation here!
    }
}
```

So the code I use is 

https://github.com/pingcap/log/pull/36/files#diff-973afd7a3fa6644d1e62a8846c6ee7dde87077ff5c0ee193624f98154eb77687R138-R148

It's **zero allocation** compared to the above one, the cost is that general duration for timeout is not supported.

Thus, the configuration here is ... several seconds, if the user configure it to "500ms", the code can't work well.
And how to express "no timeout"? `0s`? or a infinite long duration?

Reply:
I see, thanks.